pkg/controller/attach_detach.go

```diff
@@ -264,7 +265,15 @@ func (r *ReconcileAttachDetach) triggerDetach(ctx context.Context, azVolumeAttac
 				}
 			}
 
-			_ = azureutils.UpdateCRIWithRetry(updateCtx, nil, r.client, r.azVolumeClient, azVolumeAttachment, updateFunc, consts.ForcedUpdateMaxNetRetry)
+			errUpdate := azureutils.UpdateCRIWithRetry(updateCtx, nil, r.client, r.azVolumeClient, azVolumeAttachment, updateFunc, consts.ForcedUpdateMaxNetRetry)
+
+			if err == nil && errUpdate == nil {
+				//volume successfully detached
+				err := r.controllerSharedState.tryCreateFailedReplicas(ctx, attachdetach, r)
```

Comment:
We should try avoiding adding replica creation logic to `attach detach controller`. We can try creating failed replicas in `replica.go` instead (e.g. add a go routine at the end of `triggerGarbageCollection` method to wait for a volume be fully garbage collected before calling `tryCreateFailedReplicas`).

Reply:
Yeah, with @sunpa93 's async attach/detach changes, we may detach a replica to make room for a primary attachment. We wouldn't want to try to recreate failed replicas for other volumes in this case, so we'll need to trigger the retry from `triggerGarbageCollection` as you suggest..